src/main/java/spireMapOverhaul/zones/monsterZoo/MonsterZooZone.java

```diff
@@ -0,0 +1,119 @@
+import spireMapOverhaul.zoneInterfaces.CombatModifyingZone;
+import spireMapOverhaul.zoneInterfaces.RenderableZone;
+import spireMapOverhaul.zoneInterfaces.RewardModifyingZone;
+
+public class MonsterZooZone extends AbstractZone implements RewardModifyingZone, CombatModifyingZone, RenderableZone {
+    public static final String ID = "MonsterZoo";
+
+    public MonsterZooZone() {
+        super(ID, Icons.MONSTER, Icons.MONSTER, Icons.MONSTER, Icons.MONSTER, Icons.MONSTER); //Just having a bit of fun
+        this.width = 3;
+        this.maxWidth = 4;
+        this.height = 3;
+    }
+
+    @Override
+    public AbstractZone copy() {
+        return new MonsterZooZone();
+    }
+
+    @Override
+    public Color getColor() {
+        return Color.SALMON.cpy();
+    }
+
+    @Override
+    public void replaceRooms(Random rng) {
+        //Replace all non monster rooms with monster rooms
+        for (MapRoomNode node : this.nodes) {
+            if(node.room != null && !(MonsterRoom.class.equals(node.room.getClass()))) {
+                node.setRoom(new MonsterRoom());
+            }
+        }
+    }
+
+    @Override
+    protected boolean canIncludeTreasureRow() {
+        return false;
+    }
+
+    @Override
+    protected boolean canIncludeFinalCampfireRow() {
+        return false;
+    }
+
+    // Don't spawn strong monsters at the start of the first act
+    @Override
+    protected boolean canIncludeEarlyRows() {
+        return !(AbstractDungeon.actNum == 1);
+    }
+
+    // Increase number of cards in reward, upgrade chance and amount of gold.
+    @Override
+    public int changeNumberOfCardsInReward(int curNumCards) {
+        return curNumCards + 1;
+    }
+
+    @Override
+    public boolean allowUpgradingRareCards() { //Somebody has to use the fancy hook I made :p
+        return true;
+    }
+
+    @Override
+    public float changeCardUpgradeChance(float curChance) {
+        return curChance + 0.2f;
+    }
+
+    @Override
+    public void modifyReward(RewardItem rewardItem) {
+        if(rewardItem.type == RewardItem.RewardType.GOLD) {
+            rewardItem.incrementGold(rewardItem.goldAmt);
+        }
+    }
+
+    private static final float BG_MONSTER_MAX = 10f;
+    private float bgMonsterTimer = 4f;
+    @Override
+    public void update() {
+        if(AbstractDungeon.getCurrRoom() instanceof MonsterRoom) {
+            bgMonsterTimer -= Gdx.graphics.getRawDeltaTime();
+
+            if (bgMonsterTimer <= 0) {
+                bgMonsterTimer = BG_MONSTER_MAX * MathUtils.random(0.5f, 1f);
+                AbstractDungeon.effectList.add(new BackgroundMonsterEffect(true));
+            }
+        }
+    }
+
+    @Override
+    public void onVictory() {
+        BackgroundMonsterEffect.masterDispose();
+    }
+
+    // All monsters gain an amount of strength
+    @Override
+    public void atPreBattle() {
+        Wiz.forAllMonstersLiving(m -> Wiz.atb(new ApplyPowerAction(m, null, new StrengthPower(m, getStrAmt(m)))));
```

Comment:
Did you consider having a bigger set of potential buffs (that would be randomly picked from, either instead of or in addition to strength)? Not needed for merging this or anything, but if you want to build on the zone it's something you could add in. I think the extra variety would be interesting if monsters sometimes have metallicize or malleable or artifact.

Though at this point I'm just describing what Corrupt the Spire does with fights, which is a funny place to end up at.